src/cdk/tree/select-trigger.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {SelectionModel} from '@angular/cdk/collections';
+import {
+  Directive,
+  forwardRef,
+  Inject,
+  Input,
+  Optional,
+} from '@angular/core';
+import {CdkTree} from './tree';
+import {CdkTreeNode} from './node';
+
+/**
+ * Node selection trigger. Clicking on the trigger will change the selection.
+ */
+@Directive({
+  selector: '[cdkNodeSelectTrigger]',
+  host: {
+    '(click)': 'trigger($event)',
+  }
+})
+export class CdkNodeSelectTrigger<T> {
```

Comment:
I don't quite remember why this is baked into the tree- what's the background here?

Reply:
This can be used to change expand/collapse a tree node/subtree with a custom SelectionModel(a more general trigger for CdkNodeTrigger). Removed since it should not be included for the initial version.